src/api/app/models/bs_request.rb

```diff
@@ -62,6 +64,17 @@ def save!
     notify if new
   end
 
+  def self.find_by_number!(number)
```

Comment:
Or you name this "api" method some other way, not the standard activerecord way.